File: cpp/array/ArrayAddition.cpp
/*
 * "Do a program that reads 'N' real numbers and stores them in an array. Then:
 *  -Print all array elements;
 *  -Print the sum and the average of all elements stored in the array
 *  		--problem by nelio Alves
 */

 #include<iostream>
 #include<iomanip>

 using namespace std;

 int main() {

    int arraySize = 0;

    cout << "Please, insert how many numbers will be input: ";
    cin >> arraySize;

    double insertedNumbers[arraySize];

    //collects elements and sums them
    double arrayElementsSum = 0;
    for ( int i = 0; i < arraySize; i++ ) {
        cout << "Insert #" << i+1 << " number: ";
        cin >> insertedNumbers[i];

        arrayElementsSum += insertedNumbers[i];
    }

    double arrayElementsAverage = arrayElementsSum / arraySize;

    cout << "\nArray elements: ";
    for ( int i = 0; i < arraySize; i++ ) {
        cout << fixed << setprecision(1);
        cout << " " << insertedNumbers[i] << " ";
    }

    cout << fixed << setprecision(2);
    cout << "\nElements sum: " << arrayElementsSum;
    cout << "\nArray elements average: " << arrayElementsAverage << endl;

    return 0;
 }

File: python/array/ArrayAddition.py
'''
"Do a program that reads 'N' real numbers and stores them in an array. Then:
    -Print all array elements;
    -Print the sum and the average of all elements stored in the array"
        -- problem by nelio Alves
'''

arraySize = int(input("Insert how many numbers will be input: "))

insertedNumbers: [float] = [0 for x in range(arraySize)]

# collects numbers and sums them up
arrayElementsSum = 0
for i in range(0, arraySize):
    insertedNumbers[i] = float(input(f"Insert #{i+1} number: "))

    arrayElementsSum = arrayElementsSum + insertedNumbers[i]

arrayElementsAverage = arrayElementsSum / arraySize

print("\nArray elements: ", end='')
for i in range(0, arraySize):
    print(f" {insertedNumbers[i]:.1f} ", end='')

print(f"\nElements sum: {arrayElementsSum:.2f}")
print(f"Array elements average: {arrayElementsAverage:.2f}")
